File: solutions/python/resistor-color/3/resistor_color.py
"""Module for resistor color code calculations."""

RESISTOR_COLORS = (
    "black",
    "brown",
    "red",
    "orange",
    "yellow",
    "green",
    "blue",
    "violet",
    "grey",
    "white",
)
"""Tuple of resistor colors in order of their corresponding numeric value."""


def color_code(color):
    """Return the numeric value of a resistor color."""
    return RESISTOR_COLORS.index(color)


def colors():
    """Return a list of all resistor colors."""
    return list(RESISTOR_COLORS)

File: solutions/lua/resistor-color/1/resistor-color.lua
local colors = {
        "black", "brown", "red", "orange", "yellow", 
        "green", "blue", "violet", "grey", "white"
}

return {
  color_code = function(color)
    for index, value in ipairs(colors) do
      if value == color then
        return index - 1
      end
    end
  end
}
